byway/src/main/webapp/destinations.js

```diff
@@ -0,0 +1,197 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//import {ProgressBar} from  './script'
+/*
+
+
+/* exported initAutocomplete, getCurrentAddress */
+/* global google */
+
+
+const defaultCenter = {
+  lat: 40.712776,
+  lng:-74.005974
+};
+Object.freeze(defaultCenter);
+let userlatlng = {lat:null , lng: null};
+
+/*
+* Creates map and search boxes with autocomplete
+*/
+function initAutocomplete() {   
+  const map = new google.maps.Map(document.getElementById("map"), {
+    center: { lat: defaultCenter.lat, lng: defaultCenter.lng },
+    zoom: 13,
+    mapTypeId: "roadmap"
+  });
+
+  //navigator is an HTML geolocation API variable to get information about the users current location
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function(position) {
+      userlatlng = {
+        lat: position.coords.latitude,
+        lng: position.coords.longitude
+      };
+      map.setCenter(userlatlng);
+    }, function() {
+         map.setCenter(defaultCenter);
+        });
+  } 
+  else {
+    window.alert("Geolocation failed");
+  }
+  // Create the search boxes and link them to the UI elements.
+  const START_SEARCH_BOX = 'start-search-box';
+  const DESTINATIONS_SEARCH_BOX= 'destinations-search-box';
+  createSearchBox(map,START_SEARCH_BOX);
+  createSearchBox(map,DESTINATIONS_SEARCH_BOX);
+  
+}
+
+/*
+ * Creates a search box
+ * @param {google.maps.Map} map
+ * @param {string} container
+ */
+function createSearchBox(map,container){
+  const start = document.getElementById(container);
+  const searchBox = new google.maps.places.SearchBox(start);
+
+   map.addListener("bounds_changed", () => {
+    searchBox.setBounds(map.getBounds());
+  });
+  let markers = [];
+
+  searchBox.addListener("places_changed", () => {
```

Comment:
I think it would be cleaner and easier to read if you moved this anonymous function to a named function that is called. Something like `drawSearchBox` or `updateSearchBox`, whatever you feel is most descriptive.

Reply:
added it